core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -126,10 +126,10 @@ public ParsedQuery parse(QueryBuilder queryBuilder) {
             BytesReference bytes = queryBuilder.buildAsBytes();
             parser = XContentFactory.xContent(bytes).createParser(bytes);
             return parse(cache.get(), parser);
-        } catch (QueryParsingException e) {
+        } catch (QueryCreationException e) {
```

Comment:
This particular method seems only to be used in tests at the moment. I'm unsure about what all the parse() methods in IndexQueryParserService are doing since I haven't directly used them yet. Do you think those will be calles as part of the query generation or are they also called in fromXContent? Need to investigate this.